### DISCUSSION TITLE
Power Apps 질문

### DISCUSSION THREAD

**Discussion by aramorugeta** [fusiondevkr/pl900#75] [1 Q&A] [answered]
평소 Power Apps 를 사용하다 궁금한점이 생겨서 질문합니다.

1. PowerApps Video Player로 OneDrive 또는 SharePoint 내의 Video File URL을 이용하여 스트리밍이 가능할까요? (Stream의 경우 계정 문제로 게스트가 사용할 수 없어서요)

2. PowerApps 버튼으로 시작되는 PowerAutomate 흐름에서 KaKao API를 사용하여 응답받은 JSON Data 가 PowerApps 변수로 전달이 안되던데 혹시 이유를 알 수 있을까요?(스키마는 정상입니다. PowerAutomate 흐름에서는 정상적으로 Data가 받아집니다.) 링크(https://powerusers.microsoft.com/t5/Webinars-and-Video-Gallery/PowerApp-Barcode-Scanning-books-Google-Books-API/td-p/163918)와 같이 했는데 마지막 부분에서 변수에 아무것도 없습니다.

3. 응답자의 답변에 괄호"()" 가 들어가면 스키마 때문인지 JSON Parsing 할때 문제가 생깁니다. 혹시 우회하는 방법 있을까요?

**Comment by justinyoo** [comment, marked as answer]
> 1. PowerApps Video Player로 OneDrive 또는 SharePoint 내의 Video File URL을 이용하여 스트리밍이 가능할까요? (Stream의 경우 계정 문제로 게스트가 사용할 수 없어서요)

현재 비디오 관련 콘트롤은 두가지가 있습니다.

* [Power Apps의 오디오 및 비디오](https://docs.microsoft.com/ko-kr/powerapps/maker/canvas-apps/controls/control-audio-video?WT.mc_id=github-0000-juyoo)
* [Power Apps의 Microsoft Stream 비디오 컨트롤](https://docs.microsoft.com/ko-kr/powerapps/maker/canvas-apps/controls/control-stream-video?WT.mc_id=github-0000-juyoo)

해당 동영상 파일 혹은 폴더에 권한만 충분하다면 가능할 것으로 보입니다. 이 두 문서를 참조해 보세요. 저도 좀 더 찾아보겠습니다.

**Reply by aramorugeta** [reply]
확실한 방법은 없군요^^ //저도 더 찾아보겠습니다.

**Comment by justinyoo** [comment]
> 2. PowerApps 버튼으로 시작되는 PowerAutomate 흐름에서 KaKao API를 사용하여 응답받은 JSON Data 가 PowerApps 변수로 전달이 안되던데 혹시 이유를 알 수 있을까요?(스키마는 정상입니다. PowerAutomate 흐름에서는 정상적으로 Data가 받아집니다.) 링크([https://powerusers.microsoft.com/t5/Webinars-and-Video-Gallery/PowerApp-Barcode-Scanning-books-Google-Books-API/td-p/163918)와](https://powerusers.microsoft.com/t5/Webinars-and-Video-Gallery/PowerApp-Barcode-Scanning-books-Google-Books-API/td-p/163918)%EC%99%80) 같이 했는데 마지막 부분에서 변수에 아무것도 없습니다.

질문이 이해가 가지 않습니다. 파워 오토메이트에서 HTTP 액션으로 Kakao API를 호출해서 결과를 제대로 받았다면 이를 응답 액션을 통해 파워 앱으로 돌려줄 수 있어야 합니다. 만약 파워 앱에서 이 값을 못 받았다면 파워 오토메이트의 응답 액션 설정에서 오류가 있을 가능성이 높습니다. 혹시 파워 오토메이트에서 응답 액션을 어떻게 처리하셨는지 스크린샷 같은 것으로 공유해 주실 수 있으신가요?

**Reply by aramorugeta** [reply]
아래와 같이 했습니다. 스키마도 정상이고 질문에 있는 링크랑 같이 했는데 PowerApps에 변수로 전달이 안됩니다.

![Screenshot_20211029_190811](https://user-images.githubusercontent.com/49345906/139417619-c577328d-097e-4f7f-9f1d-7fa307aa443e.png)

**Reply by justinyoo** [reply]
Response 액션의 Body 항목에는 뭐가 들어가나요? 지금 보이는 것만으로는 아무것도 없어 보입니다.

**Reply by aramorugeta** [reply]
죄송합니다 수정 중 스크린 샷이었네요// 아래와 같습니다.

![Screenshot_20211030_100502](https://user-images.githubusercontent.com/49345906/139514855-50d48831-3c8a-4a9d-86ed-b81b0591454c.png)

**Comment by justinyoo** [comment]
> 3. 응답자의 답변에 괄호"()" 가 들어가면 스키마 때문인지 JSON Parsing 할때 문제가 생깁니다. 혹시 우회하는 방법 있을까요?

스키마에서 괄호(`(` `)`)는 크게 문제가 없어야 정상이긴 한데요, 혹시 샘플 데이터와 사용하셨던 JSON 스키마를 공유해 주시면 저도 테스트를 해 보겠습니다.

**Reply by aramorugeta** [reply]
스키마는 아래와 같습니다.

```json
{
    "type": "array",
    "items": {
        "type": "object",
        "properties": {
            "name": {
                "type": "string"
            },
            "link": {
                "type": "string"
            },
            "id": {
                "type": "string"
            },
            "type": {},
            "size": {
                "type": "integer"
            },
            "referenceId": {
                "type": "string"
            },
            "driveId": {
                "type": "string"
            },
            "status": {
                "type": "integer"
            },
            "uploadSessionUrl": {}
        },
        "required": [
            "name",
            "link",
            "id",
            "type",
            "size",
            "referenceId",
            "driveId",
            "status",
            "uploadSessionUrl"
        ]
    }
}
```

그리고 사용한 흐름은 아래와 같은데 왠지 답변 읽고 나니 아래 과정에서 문제가 생긴 것 같기도 합니다.(ms form으로 영상을 받았고, 공통적으로 영상 이름에 ()가 들어갑니다. ex) 영상(홍길동).mp4
![Screenshot_20211029_191509](https://user-images.githubusercontent.com/49345906/139418685-f7610ebf-eed6-4ac6-94d0-b8a1d0aab2bd.png)

**Reply by justinyoo** [reply]
스키마 자체가 유효하지 않은 것 처럼 보이는데요... 혹시 https://www.jsonschemavalidator.net/ 여기에서 스키마와 샘플 데이타를 넣어 보시겠어요?

**Reply by aramorugeta** [reply]
스키마는 문제 없었는데 아래와 같은 상황이 만들어집니다. 예상했던대로 폴더의 파일찾기에서 문제가 생겼습니다. outputs이 없습니다.
![Screenshot_20211030_102048](https://user-images.githubusercontent.com/49345906/139516083-c73c7a1f-1d52-47d2-a1cc-35ea4fdd0b92.png)
